packages/tupaia-web/src/pages/Project.tsx

```diff
@@ -17,11 +22,22 @@ const Container = styled.div`
  * This is the layout for the project/* view. This contains the map and the sidebar, as well as any overlays that are not auth overlays (i.e. not needed in landing pages)
  */
 export const Project = () => {
-  // Use these to fetch the project and any other entity info you might need
-  // const { projectCode, entityCode, '*': dashboardCode } = useParams();
+  const { projectCode } = useParams();
+  const { data: project, isLoading } = useProject(projectCode);
+  const { isLoggedIn } = useUser();
+  const { navigateToModal } = useModal();
 
+  // check if the user is logged in. If they are not logged in and the project requires login, direct first to the login modal
+  useEffect(() => {
```

Comment:
This looks good. We might have to do something similar for the entity requests and the dashboard requests so it might make sense to make something generic and I'm not sure if it's best placed in the query hook or in the view code. I think this is good for now but let's keep it in mind as we add those other use cases.